lms/views/decorators/h_api.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+
+"""View decorators for working with the h API."""
+
+from __future__ import unicode_literals
+
+import json
+import requests
+
+from pyramid.httpexceptions import HTTPBadGateway
+from pyramid.httpexceptions import HTTPBadRequest
+from pyramid.httpexceptions import HTTPGatewayTimeout
+
+from lms import util
+from lms.util import MissingToolConsumerIntanceGUIDError
+from lms.util import MissingUserIDError
+
+
+def create_h_user(wrapped):  # noqa: MC0001
+    """
+    Create a user in h if one doesn't exist already.
+
+    Call the h API to create a user for the authorized LTI user, if one doesn't
+    exist already.
+    """
+    def wrapper(request, jwt):  # pylint: disable=too-many-branches
```

Comment:
The `jwt` param here is unused, other than to pass it in to the wrapped view. This tight coupling between view decorators and between views and view decorators, with having to pass params through, is a result of how AJ are using Python decorators on views here and how they've factored the code. Needs to be refactored later.